core/src/main/java/io/questdb/log/LogAlertSocketWriter.java

```diff
@@ -0,0 +1,303 @@
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.log;
+
+import io.questdb.VisibleForTesting;
+import io.questdb.mp.QueueConsumer;
+import io.questdb.mp.RingQueue;
+import io.questdb.mp.SCSequence;
+import io.questdb.mp.SynchronizedJob;
+import io.questdb.std.*;
+import io.questdb.std.datetime.microtime.MicrosecondClock;
+import io.questdb.std.datetime.microtime.MicrosecondClockImpl;
+import io.questdb.std.str.DirectByteCharSequence;
+import io.questdb.std.str.Path;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.*;
+import java.util.function.Consumer;
+
+public class LogAlertSocketWriter extends SynchronizedJob implements Closeable, LogWriter {
+
+    static final String DEFAULT_ALERT_TPT_FILE = "/alert-manager-tpt.json";
+
+    private static final String DEFAULT_ENV_VALUE = "GLOBAL";
+    private static final String ORG_ID_ENV = "ORGID";
+    private static final String NAMESPACE_ENV = "NAMESPACE";
+    private static final String CLUSTER_ENV = "CLUSTER_NAME";
+    private static final String INSTANCE_ENV = "INSTANCE_NAME";
+    private static final String MESSAGE_ENV = "ALERT_MESSAGE";
+    private static final String MESSAGE_ENV_VALUE = "${" + MESSAGE_ENV + "}";
+
+
+    private final int level;
+    private final MicrosecondClock clock;
+    private final FilesFacade ff;
+    private final SCSequence writeSequence;
+    private final RingQueue<LogRecordSink> alertsSourceQueue;
+    private final QueueConsumer<LogRecordSink> alertsProcessor = this::onLogRecord;
+    private final DollarExpr dollar$ = new DollarExpr();
+    private final CharSequenceObjHashMap<CharSequence> alertProps = DollarExpr.adaptMap(System.getenv());
+    private HttpLogAlertBuilder alertBuilder;
+    private LogAlertSocket socket;
+
+    {
+        if (!alertProps.contains(ORG_ID_ENV)) {
+            alertProps.put(ORG_ID_ENV, DEFAULT_ENV_VALUE);
+        }
+        if (!alertProps.contains(NAMESPACE_ENV)) {
+            alertProps.put(NAMESPACE_ENV, DEFAULT_ENV_VALUE);
+        }
+        if (!alertProps.contains(CLUSTER_ENV)) {
+            alertProps.put(CLUSTER_ENV, DEFAULT_ENV_VALUE);
+        }
+        if (!alertProps.contains(INSTANCE_ENV)) {
+            alertProps.put(INSTANCE_ENV, DEFAULT_ENV_VALUE);
+        }
+        alertProps.put(MESSAGE_ENV, MESSAGE_ENV_VALUE);
+    }
```

Comment:
`TemplateParser` expects  the template to contain at the very least something like this:

```
[${ALERT_MESSAGE}]
```

It could contain something as verbose as this:

```
[
  {
    "Status": "firing",
    "Labels": {
      "alertname": "QuestDbInstanceLogs",
      "category": "application-logs",
      "severity": "critical",
      "orgid": "${ORGID}",
      "service": "QuestDB",
      "namespace": "${NAMESPACE}",
      "cluster": "${CLUSTER_NAME}",
      "instance": "${INSTANCE_NAME}"
    },
    "Annotations": {
      "description": "ERROR/${ORGID}/${NAMESPACE}/${CLUSTER_NAME}/${INSTANCE_NAME}",
      "message": "${ALERT_MESSAGE}"
    }
  }
]
```

`LogAlertSocketWriter` uses the `TemplateParser` to first parse the original template, resolving all env vars except for `${ALERT_MESSAGE}` (which represents our variable part of the final alert message). Then it calls method `toString` on the template parser, which in effect creates a new template:

```
[
  {
    "Status": "firing",
    "Labels": {
      "alertname": "QuestDbInstanceLogs",
      "category": "application-logs",
      "severity": "critical",
      "orgid": "GLOBAL",
      "service": "QuestDB",
      "namespace": "GLOBAL",
      "cluster": "GLOBAL",
      "instance": "GLOBAL"
    },
    "Annotations": {
      "description": "ERROR/GLOBAL/GLOBAL/GLOBAL/GLOBAL",
      "message": "${ALERT_MESSAGE}"
    }
  }
]
```
This final template is used to extract the "components", which will be 3:
- static body part up until the `$` in (${ALERT_MESSAGE})
- variable vody part, which is where `${ALERT_MESSAGE}` is in this template
- static body part from the `}` (in ${ALERT_MESSAGE})